### DISCUSSION TITLE
How can i query dates within my timezone?

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#18554] [Q&A]
My timezone is currently GMT-4 and I am having query issues when comparing GMT with ISO strings... 

**Problem**: if i search for date "2023-03-29", it gives me the results of dates that are after 8:00pm (hence GMT-4). Same thing, if i search for date "2023-03-28", every date after 8:00pm is not queried. What am i doing wrong?

```js
const { date } = req.query //this value comes from an input ex "2023-03-29"

    const facturas = await prisma.numero_de_factura.findMany({
      where: {
        created_at: {
          gte: new Date(date)
          lte: new Date(
            new Date(date).setDate(new Date(date).getDate() + 1)
          )
        },
```

**Comment by ludralph** [comment]
Hi @gmcamposano 👋 

It seems like you are using local time instead of UTC time when querying the database.  Can you update your code like the example shown below

```js
const { date } = req.query;
const utcDate = new Date(date).toISOString();

const facturas = await prisma.numero_de_factura.findMany({
  where: {
    created_at: {
      gte: new Date(utcDate),
      lte: new Date(new Date(utcDate).setDate(new Date(utcDate).getDate() + 1))
    }
  }
});
```

**Reply by gmcamposano** [reply]
Hello. This did not work. 
What made it work (and i really don't know why) is:
```
const fechaISO = new Date(fecha).setHours(new Date(fecha).getHours() + 4)

const facturas = await prisma.numero_de_factura.findMany({...
})
```

**Reply by ludralph** [reply]
Thanks for sharing what worked for you. I'll go ahead and close this discussion.
